### ISSUE TITLE
Cannot run on Zinc streams

### ISSUE THREAD

**Issue by jecisc** [kursjan/petitparser2#55]
In the doc it is written that PP now works on stream. I tried to parse a file but it failed. I suppose it is because file stream implementation of Pharo changed to use Zinc now. 

```
ZnCharacterReadStream(Object)>>doesNotUnderstand: #asPetit2Stream
Message>>sentTo:
ZnCharacterReadStream(Object)>>doesNotUnderstand: #asPetit2Stream
PP2ActionNode(PP2Node)>>parseAdaptable:
PP2ActionNode(PP2Node)>>parseContext:
PP2ActionNode(PP2Node)>>parse:

```

**Comment by kursjan** [comment]
This should definitely work, I will look into this asap.

**Comment by kursjan** [comment]
Indeed, there is no extension for `ZnCharacaterStream`. But you can build your own stream on top of ZincStream:

	| byteStream stream |
	byteStream := ZnClient new
 		url: 'http://pharo.org';
  		streaming: true;
  		get.
	stream := PP2CharacterStream on: byteStream encoder: ZnUTF8Encoder new.

	^ PP2HtmlHeaderGrammar new optimize parse: stream


See `PP2HtmlHeader class>>example`

**Comment by kursjan** [comment]
Which version of Pharo are you using? It seems there is an implementation of asPetit2Stream in Pharo 6.1:

```smalltalk
ZnCharacterReadStream>>asPetit2Stream
	^ PP2BufferStream on: self
```

As a workaround, consider creating your own PP2BufferStream on ZnCharacterReadStream by simply wrapping to `PP2BufferStream`.

**Comment by kursjan** [comment]
I also improved the documentation of PP2 related streams in https://github.com/kursjan/petitparser2/commit/1e20acad0febb34a798e9ac44404a83d300b17c9

**Comment by jecisc** [comment]
I am on Pharo 8. 

In the end I cannot use this feature but I was repporting it in case you want to add syntactic suggar.
